### PR TITLE
template: remove templateprocessing dep from new-app

### PR DESCRIPTION
**Pull request by mfojtik**
Remove `templateprocessing` dependency from new-app, making `templateprocessing` only depend on legacygroupification.

@soltysh i think this helps you.

### CODE REVIEW

File: pkg/template/templateprocessing/template.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	appsv1 "github.com/openshift/api/apps/v1"
 	templatev1 "github.com/openshift/api/template/v1"
 	. "github.com/openshift/library-go/pkg/template/generator"
+
 	"github.com/openshift/origin/pkg/api/legacygroupification"
```
Comment (mfojtik) [inline pkg/template/templateprocessing/template.go]: @deads2k ^ this thing exists to deal with legacy templates that has non-groupified resources... we discussed with @soltysh whether it makes sense to move it to `openshift/api`.

File: pkg/template/templateprocessing/template.go [hunk 2]
```diff
@@ -28,7 +29,7 @@ type Processor struct {
 	Generators map[string]Generator
 }
 
-// NewProcessor creates new Processor and initializes its set of generators.
+// NewProcessor creates new Processor and initializv1es its set of generators.
```
Comment (deads2k) [inline pkg/template/templateprocessing/template.go]: how did you typo this.... ? :)

Comment (mfojtik) [reply pkg/template/templateprocessing/template.go]: lol :-)

**Pull request merged**